test/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+project(Template_test)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
+set(external_libs)
+if (NOT Template_IS_TOP_LEVEL)
+	FetchContent_Declare(Template
+			GIT_REPOSITORY https://github.com/LecrisUT/CMake-Template
+			GIT_TAG main
+			FIND_PACKAGE_ARGS CONFIG
+	)
+	list(APPEND external_libs Template)
+endif ()
```

Comment:
This structure I think it is good to advertise. This allows for importing and running the tests individually from an external testing suite like `tmt` or from a cross-compiled environment like in `conda`.